### PR TITLE
Move wifi hotspot control to patchbox-wifi

### PR DESCRIPTION
**Pull request by ozbenh**
Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>

### CODE REVIEW

**Comment by gtrainavicius** [comment]
I've made a couple of changes, but looks like the permission to allow my commits is not enabled :) https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork

Anyway, here's a patch:

```
pi@raspberrypi:/usr/local/pisound/scripts/pisound-btn $ git format-patch -1 HEAD
0001-Restore-original-wifi-hotspot-is-active-check-fixed-.patch
pi@raspberrypi:/usr/local/pisound/scripts/pisound-btn $ cat 0001-Restore-original-wifi-hotspot-is-active-check-fixed-.patch
From aa0fe19cd60cefe1bea0ac3de1d4f55c0fdeb694 Mon Sep 17 00:00:00 2001
From: Giedrius <giedrius@blokas.io>
Date: Thu, 4 Nov 2021 13:36:07 +0000
Subject: [PATCH] Restore original wifi-hotspot is-active check, fixed [[ not
 found error in disable_wifi_hotspot.sh

---
 scripts/pisound-btn/disable_wifi_hotspot.sh |  2 +-
 scripts/pisound-btn/toggle_wifi_hotspot.sh  |  4 +---
 scripts/pisound-btn/wifi_hotspot_status.sh  | 25 ---------------------
 3 files changed, 2 insertions(+), 29 deletions(-)
 delete mode 100755 scripts/pisound-btn/wifi_hotspot_status.sh

diff --git a/scripts/pisound-btn/disable_wifi_hotspot.sh b/scripts/pisound-btn/disable_wifi_hotspot.sh
index 506e48b..ab540ad 100755
--- a/scripts/pisound-btn/disable_wifi_hotspot.sh
+++ b/scripts/pisound-btn/disable_wifi_hotspot.sh
@@ -20,7 +20,7 @@

 . /usr/local/pisound/scripts/common/common.sh

-if [[ "$(systemctl is-system-running || true)" == "stopping" ]]; then
+if [ "$(systemctl is-system-running || true)" = "stopping" ]; then
        flash_leds 1
        log "System is stopping anyway, skipping disabling WiFi hotspot."
        exit
diff --git a/scripts/pisound-btn/toggle_wifi_hotspot.sh b/scripts/pisound-btn/toggle_wifi_hotspot.sh
index a8b402b..1f09867 100755
--- a/scripts/pisound-btn/toggle_wifi_hotspot.sh
+++ b/scripts/pisound-btn/toggle_wifi_hotspot.sh
@@ -24,9 +24,7 @@ SCRIPTS_PATH="/usr/local/pisound/scripts"

 log "Toggling WiFi hotspot!"

-STATUS=$(${SCRIPTS_PATH}/pisound-btn/wifi_hotspot_status.sh)
-
-if [ "$STATUS" = "enabled" ]; then
+if systemctl is-active wifi-hotspot --quiet; then
        ${SCRIPTS_PATH}/pisound-btn/disable_wifi_hotspot.sh
 else
        ${SCRIPTS_PATH}/pisound-btn/enable_wifi_hotspot.sh
diff --git a/scripts/pisound-btn/wifi_hotspot_status.sh b/scripts/pisound-btn/wifi_hotspot_status.sh
deleted file mode 100755
index df18100..0000000
--- a/scripts/pisound-btn/wifi_hotspot_status.sh
+++ /dev/null
@@ -1,25 +0,0 @@
-#!/bin/sh
-
-# pisound-btn daemon for the Pisound button.
-# Copyright (C) 2017  Vilniaus Blokas UAB, https://blokas.io/pisound
-#
-# This program is free software; you can redistribute it and/or
-# modify it under the terms of the GNU General Public License
-# as published by the Free Software Foundation; version 2 of the
-# License.
-#
-# This program is distributed in the hope that it will be useful,
-# but WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-# GNU General Public License for more details.
-#
-# You should have received a copy of the GNU General Public License
-# along with this program; if not, write to the Free Software
-# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
-#
-
-if systemctl is-active wifi-hotspot --quiet; then
-    echo "enabled"
-else
-    echo "disabled"
-fi
--
2.20.1

```

File: pisound-btn/debian/DEBIAN/control [hunk 1]
```diff
@@ -1,10 +1,9 @@
 Package: pisound-btn
-Version: 1.14-1
+Version: 1.15-1
 Section: sound
 Priority: optional
 Architecture: armhf
-Pre-Depends: hostapd (>= 2:2.6), dnsmasq (>= 2.72)
-Depends: libc6 (>= 2.19), libgcc1 (>= 1:4.9.2-10), python3 (>= 3.5), python3-urwid (>= 1.0), git (>= 1:2.1.4), libglib2.0-bin (>= 2.33.12)
+Depends: libc6 (>= 2.19), libgcc1 (>= 1:4.9.2-10), python3 (>= 3.5), python3-urwid (>= 1.0), git (>= 1:2.1.4), libglib2.0-bin (>= 2.33.12), patchbox-wifi, patchbox-cli
```
Comment (gtrainavicius) [inline pisound-btn/debian/DEBIAN/control]: patchbox-cli dependency should not be necessary, and it's undesired for pisound-btn to depend on it - some users may just want pisound-btn installed, without much dependencies of patchbox-os. (wifi depency is fine, as it's minimal)

Also I think it's a good idea to always set the >= for the version, I remember having some difficulties caused by not having it when releasing upgrades, but I can't recall the details anymore...

Comment (ozbenh) [reply pisound-btn/debian/DEBIAN/control]: Yup. Will fix. Note: I do have "Allow edits by maintainers" checked...

**Comment by ozbenh** [comment]
Pushed with your changes and the above fix

**Pull request merged**